### ISSUE TITLE
Field sort order can be indeterminate 

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18542]
When we return the list of fields on a content type, we only order by `sort_order`.  We have a bunch of content types that have indeterminate sort orders in their - like the host type below:

```
dotcms=# select field_name, velocity_var_name, sort_order from field where structure_inode='855a2d72-f2f3-4169-8b04-ac5157c4380c';
       field_name        | velocity_var_name | sort_order
-------------------------+-------------------+------------
 fields-0                | fields0           |          1
 fields-1                | fields1           |          1
 Host Name               | hostName          |          1
 Aliases                 | aliases           |          1
 Tag Storage             | tagStorage        |          1
 Is Default              | isDefault         |          2
 Is System Host          | isSystemHost      |          3
 Host Thumbnail          | hostThumbnail     |          4
 Run Dashboard           | runDashboard      |          5
 Meta Data (Default)     | metaData          |          6
 Keywords                | keywords          |          7
 Description             | description       |          8
 Site Tags               | siteKeys          |          9
 Google Map              | googleMap         |         10
 Google Analytics        | googleAnalytics   |         12
 Add This                | addThis           |         14
 Proxy Url for Edit Mode | proxyEditModeUrl  |         17
 Embedded Dashboard Url  | embeddedDashboard |         18
```

This means when someone pushes a content type from one server to another, the sort orders can display completely differently based on the database and as soon as the content type is persisted, it will appear to have been reordered.

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/18543

**Comment by nollymar** [comment]
**Passed Internal QA (note to QA)** : The best way to test the fix was through ITs. However, it is recommended to use the old starter to verify that the sort order is working correctly.

**Comment by bryanboza** [comment]
Fixed, tested on release-5.3.2 // Postgres and now this appear in the correct order 
![image](https://user-images.githubusercontent.com/2641437/83818989-ede09100-a685-11ea-865a-31aa620beeea.png)

**Comment by jcastro-dotcms** [comment]
@bryanboza why are the numbers 5 and 6 duplicated in the list?
